Cleanup code - st55392 Karel Andres

kar-los97 opened the pull request:
Přejmenování tříd Cplx -> ComplexNumber, Poly -> Polynome a jejich některých parametrů.
Přesun logiky z třídy Main do vlastní třídy NewtonFractal a rozdělení do více kratších metod.
Přejmenování některých proměnných a metod -> zvolil jsem výstižnější název (dle toho co metoda dělá a k čemu slouží proměnná).

Commit:
Přesun tříd Cplx, Poly do samostatných tříd.
Přejmenování tříd Cplx -> ComplexNumber, Poly -> Polynome a jejich některých parametrů.
Přesun logiky z třídy Main do vlastní třídy NewtonFractal a rozdělení do více kratších metod.
Přejmenování některých proměnných a metod -> zvolil jsem výstižnější název (dle toho co metoda dělá a k čemu slouží proměnná).

Commit:
Formátování kódu.
Rozdělení testů do samostatných tříd + přidání dalších testů.

rodi0878 commented on INPTPZ1/Mathematics/ComplexNumber.cs:
```diff
@@ -0,0 +1,74 @@
+    public class ComplexNumber
+    {
+        public double Real { get; set; }
+        public double Imaginari { get; set; }
```
ImaginarI???

rodi0878 commented on INPTPZ1/Mathematics/ComplexNumber.cs:
```diff
@@ -0,0 +1,74 @@
+        {
+            if (obj is ComplexNumber)
+            {
+                ComplexNumber x = obj as ComplexNumber;
```
x?

rodi0878 commented on INPTPZ1/Mathematics/ComplexNumber.cs:
```diff
@@ -0,0 +1,74 @@
+            }
+            return base.Equals(obj);
+        }
+        public readonly static ComplexNumber Zero = new ComplexNumber()
```
constant not at the top of class

rodi0878 commented on INPTPZ1/Mathematics/Polynome.cs:
```diff
@@ -0,0 +1,105 @@
+            Coeficients = new List<ComplexNumber>();
+        }
+
+        public void Add(ComplexNumber coe)
```
missing doc comment, please use them consistently in all project

rodi0878 commented on INPTPZ1/NewtonFractal.cs:
```diff
@@ -0,0 +1,152 @@
+
+        public NewtonFractal(string[] args)
+        {
+            polynome = new Polynome();
```
extract polynome creation code to method

rodi0878 commented on INPTPZ1/NewtonFractal.cs:
```diff
@@ -0,0 +1,152 @@
+
+        private void ColorizeBmpOutput()
+        {
+            double xmin = doubleargs[0];
```
method is named "ColorizeBmpOutput" and not "parse args"

rodi0878 commented on INPTPZ1/NewtonFractal.cs:
```diff
@@ -0,0 +1,152 @@
+            double xstep = (xmax - xmin) / intargs[0];
+            double ystep = (ymax - ymin) / intargs[1];
+
+            for (int i = 0; i < intargs[0]; i++)
```
dont like "intargs[0]" please refactor to something more meaningful

rodi0878 commented on INPTPZ1/NewtonFractal.cs:
```diff
@@ -0,0 +1,152 @@
+                    double y = ymin + i * ystep;
+                    double x = xmin + j * xstep;
+
+                    ox = new ComplexNumber()
```
extract creation of ox and settings of its arguments to non-zero values

rodi0878 commented on INPTPZ1/NewtonFractal.cs:
```diff
@@ -0,0 +1,152 @@
+        private int FindSolutionRootNumber()
+        {
+            var known = false;
+            var id = 0;
```
this should be index not id

rodi0878 commented on INPTPZ1/NewtonFractal.cs:
```diff
@@ -0,0 +1,152 @@
+
+        private void ColorizePixel(int id, int it, int i, int j)
+        {
+            Color vv = colors[id % colors.Length];
```
vv?

rodi0878 commented on INPTPZ1/NewtonFractal.cs:
```diff
@@ -0,0 +1,152 @@
+        private void ColorizePixel(int id, int it, int i, int j)
+        {
+            Color vv = colors[id % colors.Length];
+            vv = Color.FromArgb(vv.R, vv.G, vv.B);
```
this row doesnt do anything

rodi0878 reviewed (changes requested):
Please look at mentioned problems and try to fix them

kar-los97 commented:
Problémy opraveny

rodi0878 commented:
Možná lokálně u Vás, ale na githubu není žádný nový commit k dispozici...

Commit:
Oprava kódu

kar-los97 commented:
Omlouvám se, neprovedl jsem push. Teď už by měli na gitu být všechny změny.

rodi0878 commented:
Projekt stále obsahuje celou řadu nedostatku:
- pořadí metod v ComplexNumber je nerozumné, bylo by vhodné metody logicky (tematicky) seřadit
- Polynome::Add(coe) - coe?
- Polynome::Evaluate(x) - zbytečné vytváření dočasné proměnné
- NewtonFractal::CreateComplexNumberAndEditZeroParams - mix double a float konstanty
- NewtonFractal::ColorizeBmpOutput - nepřesný název, co to je "ox", "id" (žádné ID se v programu nikde nepoužívá)
- NewtonFractal::DoNewtonMethod - literály 30, 0.5
- NewtonFractal::FindSOlutionRootNumber - zbytečně neefektivní algoritmus procházející celý cyklus

Commit:
ComplexNumber - Úprava pořadí metod
Polynome - přejmenování parametru metody Add, úprava metody Evaluate
NewtonFractal - úprava konstant v metodě CreateComplexNumberAndEditZeroParams

kar-los97 commented:
Úpravy provedeny

rodi0878 commented:
Akceptováno
